diff --git c/bin/start-stdio w/bin/start-stdio
new file mode 100755
index 0000000..7f95ae1
--- /dev/null
+++ w/bin/start-stdio
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+# used for mason.nvim
+
+cd "$(dirname "$0")"/.. || exit 1
+
+./bin/start --stdio "$0"
diff --git c/lib/refactorex/application.ex w/lib/refactorex/application.ex
index b726e4b..76d4f12 100644
--- c/lib/refactorex/application.ex
+++ w/lib/refactorex/application.ex
@@ -9,7 +9,7 @@ defmodule Refactorex.Application do
       [
         {
           GenLSP.Buffer,
-          [communication: {GenLSP.Communication.TCP, [port: port()]}]
+          [communication: communication_config()]
         },
         {Refactorex.Logger, []},
         {Refactorex.NameCache, []},
@@ -20,10 +20,24 @@ defmodule Refactorex.Application do
     )
   end
 
-  defp port do
+  defp parse_opts do
     System.argv()
-    |> OptionParser.parse(strict: [port: :integer])
+    |> OptionParser.parse(strict: [port: :integer, stdio: :boolean])
     |> elem(0)
+  end
+
+  defp port do
+    parse_opts()
     |> Keyword.get(:port, @default_port)
   end
+
+  defp communication_config do
+    opts = parse_opts()
+
+    if Keyword.get(opts, :stdio, false) do
+      {GenLSP.Communication.Stdio, []}
+    else
+      {GenLSP.Communication.TCP, [port: port()]}
+    end
+  end
 end
